src/Microsoft.Tye.Core/KubernetesManifestGenerator.cs

```diff
@@ -452,6 +462,64 @@ internal static class KubernetesManifestGenerator
             return new KubernetesDeploymentOutput(project.Name, new YamlDocument(root));
         }
 
+        private static void AddProbe(ServiceBuilder service, YamlMappingNode container, ProbeBuilder builder, string name)
+        {
+            var probe = new YamlMappingNode();
+            container.Add(name, probe);
+
+            if (builder.Http != null)
+            {
+                var builderHttp = builder.Http!;
+                var httpGet = new YamlMappingNode();
+
+                probe.Add("httpGet", httpGet);
+                httpGet.Add("path", builderHttp.Path);
+
+                if (builderHttp.Protocol != null)
+                {
+                    httpGet.Add("scheme", builderHttp.Protocol!.ToUpper());
+                }
+
+                if (builderHttp.Port != null)
+                {
+                    httpGet.Add("port", builderHttp.Port!.ToString()!);
+                }
+                else
+                {
+                    // If port is not given, we pull default port
+                    var binding = service.Bindings.First(b => builderHttp.Protocol is null || b.Protocol == builderHttp.Protocol);
```

Comment:
It does feel a bit awkward to have health checks per service rather than per endpoint, but it seems to be what k8s does. Or maybe we require a port?

Reply:
requiring a port will indeed make it more compatible with k8s, but it will make the `tye.yaml` more verbose and less maintainable in my opinion. one thing that k8s does to help with maintainability is to allow to specify a variable as a port. e.g.

```
port: liveness-port
```